src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+  const props = sourceString.split(';').map((prop) => prop.trim());
+
+  for (const prop of props) {
+    const [key, value] = prop.split(':');
+
+    if (key) {
+      obj[key.trim()] = value.trim();
+    }
+  }
+
+  return obj;
```

Comment:
we can do it in `reduce` method. Try it
just passed ur logic in the middle of the `reduce` method

```suggestion
  return sourceString.split(';').reduce((style, declaration) => {
    
    .........

    return style;
  }, {});
```